llvm/lib/Transforms/Instrumentation/HWAddressSanitizer.cpp

```diff
@@ -1897,16 +1898,16 @@ void HWAddressSanitizer::ShadowMapping::init(Triple &TargetTriple,
     InTls = false;
     Offset = 0;
     WithFrameRecord = true;
-  } else if (ClMappingOffset.getNumOccurrences() > 0) {
-    InGlobal = false;
-    InTls = false;
-    Offset = ClMappingOffset;
-    WithFrameRecord = false;
   } else if (ClEnableKhwasan || InstrumentWithCalls) {
     InGlobal = false;
     InTls = false;
-    Offset = 0;
+    Offset = optOr<uint64_t>(ClMappingOffset, (uint64_t)0);
```

Comment:
Why is this necessary? ClMappingOffset is opt<uint64_t> so I don't understand why we need the explicit specialization.

Also please don't use C-style casts.